File: PHP/Structural/decorator.php
<?php

	interface Component{
       	public function execute();
    }

    class ConcreteComponent implements Component{
       	public function execute(){
        	return 'Message';
        }
    }

    class BaseDecorator implements Component{
       	private Component $wrappee;

       	public function __construct(Component $c){
            $this->wrappee = $c;
        }

        public function execute(){
        	return $this->wrappee->execute();
        }
    }

    class ConcreteDecorator extends BaseDecorator{
    	public function execute(){
        	return $this->extra(parent::execute());
        }

        public function extra(String $data){
    		return 'Added ' . $data;
        }
    }

	$a = new ConcreteComponent();
    $b = new ConcreteDecorator($a);

    echo $b->execute();
?>

File: Python/Structural/decorator.py
import abc

# Structure
class Component(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def execute(self):
        pass

# Object to wrap
class ConcreteComponent(Component):
    def execute(self):
        print("My component do this.")

""" 
    Base Decorator to put the similar code of 
    your Concrete Decorators thought inheritance 
"""
class BaseDecorator(Component):
    def __init__(self, c: Component):
        self.wrappee = c

    def execute(self):
        self.wrappee.execute()

# The wrapper
class ConcreteDecorator(BaseDecorator):
    def execute(self):
        self.more()
        return super().execute()

    def more(self):
        print("Decorated!")


example = ConcreteDecorator(ConcreteComponent())
example.execute()